### PR TITLE
Try to stop actions running on draft PR

### LINKED ISSUE #311: Fix CI running on draft PRs

### PR DESCRIPTION
**Pull request by SJaffa**
This fix suggested [here](https://github.com/orgs/community/discussions/25722#discussioncomment-3248907)

### CODE REVIEW

**Review by tkoskela — DISMISSED** [review]

**Comment by tkoskela** [comment]
I pushed a commit, why are the actions not running? :sob: 

Does it only run it once when marked ready for review? I don't think that's the behaviour we want

**Comment by tkoskela** [comment]
Now that I look at it, the old version only has the `if` statement on pull request, but not on push. Maybe that's why it was still running on push? Do you mind if I play with it?

Could try [this suggestion](https://github.com/orgs/community/discussions/25722#discussioncomment-3248917).

**Comment by SJaffa** [comment]
> Could try [this suggestion](https://github.com/orgs/community/discussions/25722#discussioncomment-3248917).

Yeah that looks good, we want it to run the actions when we mark it as not-a-draft, that's when I think I have fixed the problem and want to test it and get a review.

**Comment by tkoskela** [comment]
> > Could try [this suggestion](https://github.com/orgs/community/discussions/25722#discussioncomment-3248917).
> 
> Yeah that looks good, we want it to run the actions when we mark it as not-a-draft, that's when I think I have fixed the problem and want to test it and get a review.

Yes, I agree we want it to run when we mark it as ready for review. But I would also like it to re-run whenever I push commits to a ready-for-review PR, to make sure the tests pass on the most recent version. That does not seem to be happening in this PR at the moment.

**Comment by SJaffa** [comment]
Ok I think this is working now...

> pull_request:
    types: [opened, synchronize, reopened, ready_for_review]

This bit makes it run whenever you 
- open a new PR, 
- push new commits to an open PR, or 
- mark a draft PR as ready for review. 
 
This would trigger on draft PRs too when they are opened or synchronised so...

> if: github.event.pull_request.draft == false

This bit skips the `build` job, which is the only job in the workflow. That means the workflow script is run so we do get the little green tick next to the commit number showing that the workflow didn't fail, but if you look in the Actions tab it shows that it was skipped and nothing actually ran.

> push:
    branches: [ development ]

This bit should make it run the workflow when the PR is merged, but I haven't tested that. I left this in the cmake action and the documentation action but removed the rest of the triggers from the docs action as I think we only want the documentation updated when we push things into `development`

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -91,6 +93,8 @@ jobs:
     - name: Install Conan
       id: conan
       uses: turtlebrowser/get-conan@main
+      with:
+        version: 1.54.0
```
Comment (tkoskela) [inline .github/workflows/cmake.yml]: Has this change gone back in by mistake from an older version

```suggestion
```

**Review by tkoskela — CHANGES_REQUESTED** [review]
Looks good, I tested it in a fork to not cause a mess in the main repo and the behaviour is what I would expect. There's two outdated lines that fix the conan version we have removed in development. It's probably best to merge development to this PR so that they don't cause conflicts.
- pushing to draft PRs -- only CMake runs, gets skipped
- marking PRs ready -- only CMake runs
- pushing to ready PRs -- only CMake runs
- merging PRs to development -- CMake and Documentation run

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (SJaffa) [reply .github/workflows/cmake.yml]: Yes I think that was me branching off from the wrong branch, I'll remove it

**Review by tkoskela — APPROVED** [review]

**Pull request merged**